include/harness/abi-cafe-rules.toml

```diff
@@ -0,0 +1,19 @@
+# i128 types are fake on windows so this is all random garbage that might
+# not even compile, but that datapoint is a little interesting/useful
+# so let's keep running them and just ignore the result for now.
+#
+# Anyone who cares about this situation more can make the expectations more precise.
+[targets.x86_64-pc-windows-msvc."i128::cc_toolchain"]
+random = true
+[targets.x86_64-pc-windows-msvc."u128::cc_toolchain"]
+random = true
+
+# FIXME: investigate why this is failing to build
+[targets.x86_64-pc-windows-msvc."EmptyStruct::cc_toolchain"]
+busted = "build"
+[targets.x86_64-pc-windows-msvc."EmptyStructInside::cc_toolchain"]
+busted = "build"
+
+# CI GCC is too old to support _Float16
+[targets.x86_64-unknown-linux-gnu."f16::conv_c"]
```

Comment:
Having some way to support arbitrary `cfg()` conditions would be really useful to indicate that something is broken on a certain architecture or OS without having to list every single target with this architecture or OS you want to test.

Reply:
Hmm yeah I agree, but am definitely running out of steam to build out this feature myself, and am hoping this is an acceptable MVP X3

A few things:

* I specifically made it so you can do `[targets."*"."blhablhablha"]` if you really don't care
* I expect most users are going to be copy-pasting what the new suggestion system spits out, so verbosity is hopefully less of an issue?
  * And having "too many" records makes it easier to granularly update them if one platform is fixed?

Reply:
Partially there is some sense of like, when bringing up a new platform, I think you *do* want some expectations tabula-rasa? But I might be wrong.

Reply:
If you have a test failing on x86_64-unknown-linux-gnu, it is almost guaranteed to fail on all x86_64 unix targets as those share a single abi document. The only cases I know of where the OS actually matters are unix vs windows-msvc vs windows-gnu(?) on x86_64 and (non-apple) unix vs apple vs windows arm64ec on arm64.

Reply:
Hmm are there any libraries y'all compiler devs have for evaling this stuff at runtime?

Reply:
You could try using the target-lexicon crate. It has a `default_calling_convention` method which returns the calling convention used for `extern "C"` (also mostly applies to `extern "Rust"`) on the respective target. It doesn't yet handle arm64ec though.